tools/nuget/Desktop/Microsoft.MixedReality.WebRTC.props

```diff
@@ -0,0 +1,9 @@
+﻿<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Condition="'$(UsingMicrosoftNETSdk)' != 'true'">
```

Comment:
non-SDK style .NET projects can use PackageReference, but it's a small percentage.

Having said that, I was looking thought [NuGet's targets file](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets), and I see that `RestoreProjectStyle` is set in the target `_GetRestoreProjectStyle`. However, I couldn't figure out how package authors (or even customers in their own props/targets, like Directory.Build.targets) can condition on this property value.

@nkolev92 @jeffkl any ideas? Should they use `CollectPackageReferences`?  I always assumed that was for PackageReference only, whereas this props file is trying to add a `Content` element only for non-PackageReference package usage.

Reply:
I don't think you can do it. 

Also this targets file likely needs to be a target framework specific location in order to avoid triggering false compatibility (build/packageid.props makes it compatible with everything.)

Reply:
@nkolev92 how do we do that? I'm really at my limit of understanding of NuGet here, barely following the conversation above. And is there an alternative then to drive a condition block on whether `PackageReference` or `packages.config` is used, that would be better than `$(UsingMicrosoftNETSdk)`?

Reply:
In the nuspec file, where it puts `Microsoft.MixedReality.WebRTC.props` in the package, instead of putting it in `build\`, put it in `build\netstandard2.0\`, since your package appears to contain only `netstandard2.0` assets.

If your package supported multiple TFMs, for example you had `lib\netstandard2.0\` and `lib\net5.0\`, then you'd need the msbuild files in both `build\netstandard2.0\` and `build\net5.0\`.

Reply:
Regarding "correctly" supporting PackageReference vs packages.config (or Paket), unfortunately we don't seem to have a solution. I don't have other ideas on how we can easily support all PackageReference and packages.config (include Paket) perfectly.

I'm guessing by the repo name that there might be significant usage from UWP apps?  My memory is that UWP projects are non-SDK style and might use PackageReference. So this PR will cause both this `<Content` to copy the file, in addition to NuGet's `PackageReference` feature, so there will be double writes of this file. I don't think there are any more serious consequences than double writing the file.

Reply:
For `PackageReference` projects the `.csproj.nuget.g.props` file defines a `$(NuGetProjectStyle)` property. 

```
<NuGetProjectStyle Condition=" '$(NuGetProjectStyle)' == '' ">PackageReference</NuGetProjectStyle>
```

Under `VS2017/VS2019` this would appear to be set for `SDK Style` projects.
I also tested under `VS/2017/2019` for `non-SDK style` project using `PackageReference` and it's also set as expected. 

I'm unfamiliar with `Paket`. Does anyone know if this would be a better check than `$(UsingMicrosoftNETSdk)' != 'true'`? 

```
<ItemGroup  Condition=" '$(NuGetProjectStyle)' == ''">
```

Is anyone aware of any circumstances where this check would be incorrect?

I'm also interested in adding better support for `non-SDK style` projects to my own project [CefSharp](https://github.com/cefsharp/CefSharp) and have been following https://github.com/NuGet/docs.microsoft.com-nuget/issues/2070#issuecomment-767706280. 

Thanks for your time.

Reply:
Good idea/find @amaitland. I'd suggest that `'$(NuGetProjectStyle)' != 'PackageReference'` would be a better condition, although I didn't test it.

` '$(NuGetProjectStyle)' == ''` will always evaluate to true in `Directory.Build.props` (but so will `!= 'PackageReference'`) because MSBuild evaluates top down, and this file is always imported before the project's `<project>.nuget.g.props`. In a project file, it will evaluate to `true` when the `obj/` folder is empty (before NuGet restore after `git clean -xdf`, or clean checkout on a CI agent). This is why it's important to use `msbuild -restore`, not `msbuild -t:restore,build` if trying to restore & build in a single command line when using msbuild.exe. The `dotnet` CLI does this automatically.

But, it should actually work really great in a package's props/targets file. If the project uses `PackageReference`, the package's props file will be imported by the file that defines this property, so it's guaranteed to be set, and set to `PackageReference`. For packages.config, the property won't be set, hence MSBuild treats it equal to the empty string. I also don't know how Paket works, but if they define `NuGetProjectStyle` to `PackageReference` then frankly it's their own fault for misrepresenting their tool. If they set it to `Paket`, then the `!= 'PackageReference'` condition will ensure it doesn't get skipped accidentally. Similarly, if NuGet starts setting `NuGetProjectStyle` to `PackagesConfig` for `packages.config` projects (if you check NuGet's `NuGet.targets` linked above, we already do this, but in a target that should not currently trigger in packages.config projects), the condition will also keep working as planned.

So yes, I recommend this based on @amaitland's suggestion:
```suggestion
  <ItemGroup Condition="'$(NuGetProjectStyle)' != 'PackageReference'">
```

Reply:
@zivkan Thanks for the additional insight 👍

Reply:
> In the nuspec file, where it puts `Microsoft.MixedReality.WebRTC.props` in the package, instead of putting it in `build\`, put it in `build\netstandard2.0\`, since your package appears to contain only `netstandard2.0` assets.
> 
> If your package supported multiple TFMs, for example you had `lib\netstandard2.0\` and `lib\net5.0\`, then you'd need the msbuild files in both `build\netstandard2.0\` and `build\net5.0\`.

Good catch @zivkan! @djee-ms  I moved the props file into the framework-specific folders. Originally I had the props files as part of the CSPROJ and I did not check the target framework when adding them to the `.nuspec` files. My bad!